compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -2646,6 +2651,13 @@ public boolean isReAtomNode(NodeKind kind) {
         }
     }
 
+    private boolean isXmlnsInDifferentCompUnit(BXMLNSSymbol symbol, BXMLNSSymbol foundSym) {
+        if (foundSym.compUnit != null && symbol.compUnit != null) {
+            return !foundSym.compUnit.value.equals(symbol.compUnit.value);
+        }
+        return foundSym.compUnit == null ^ symbol.compUnit == null;
```

Comment:
This can then be something like `foundSymModuleXmlns != symModuleXmlns`?

Reply:
Meant specifically this line where foundSymModuleXmlns is foundSym.compUnit != null and similarly symModuleXmlns is symbol.compUnit != null.